tests/test_builders/xpath_util.py

```diff
@@ -27,6 +27,17 @@ def pformat(node: Element) -> str:
     return ''.join(f'(i={index}) {pformat(node)}\n' for index, node in enumerate(nodes))
 
 
+def _intradocument_hyperlink_check(nodes: Sequence[Element]) -> None:
```

Comment:
Could you add a comment saying what does what? When I came to some of my functions I forgot about many things about them...

And I think it's better not to put it in the `xpath_util` because it's quite specific for some tests. Don't we have some `html`-only helpers?

Reply:
> And I think it's better not to put it in the `xpath_util` because it's quite specific for some tests. Don't we have some `html`-only helpers?

We don't yet, but this function is only used in a single test module currently - so I've moved it there.

Reply:
> > And I think it's better not to put it in the `xpath_util` because it's quite specific for some tests. Don't we have some `html`-only helpers?
> 
> We don't yet, but this function is only used in a single test module currently - so I've moved it there.

Nope; that's not true at all.  I'll go ahead and extract a common utility module for this after all.